crates/common_utils/src/id_type/global_id.rs

```diff
@@ -0,0 +1,72 @@
+#![allow(unused)]
+
+use error_stack::ResultExt;
+
+use crate::{
+    consts::{
+        CELL_IDENTIFIER_LENGTH, MAX_ALLOWED_MERCHANT_REFERENCE_ID_LENGTH,
+        MIN_REQUIRED_MERCHANT_REFERENCE_ID_LENGTH,
+    },
+    errors, generate_time_ordered_id,
+    id_type::LengthId,
+};
+
+#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+/// A global id that can be used to identify any entity
+/// This id will have information about the entity and cell in a distributed system architecture
+struct GlobalId(
+    LengthId<MAX_ALLOWED_MERCHANT_REFERENCE_ID_LENGTH, MIN_REQUIRED_MERCHANT_REFERENCE_ID_LENGTH>,
```

Comment:
These MAX and MIN length should be different from `MAX_ALLOWED_MERCHANT_REFERENCE_ID_LENGTH` `MIN_REQUIRED_MERCHANT_REFERENCE_ID_LENGTH`